### PR TITLE
RUST-418 Add test to verify that $readPreference isn't sent to standa…

### PR DESCRIPTION
**Pull request by saghm**
…lones

### CODE REVIEW

File: src/test/coll.rs [hunk 1]
```diff
@@ -720,3 +723,34 @@ async fn find_one_and_delete_hint_server_version() {
         assert!(res.is_ok());
     }
 }
+
+#[cfg_attr(feature = "tokio-runtime", tokio::test)]
+#[cfg_attr(feature = "async-std-runtime", async_std::test)]
+#[function_name::named]
+async fn no_read_preference_to_standalone() {
```
Comment (saghm) [inline src/test/coll.rs:730]: We already have the correct behavior required by the drivers ticket, so I implemented a test that verifies it so we don't regress

File: src/test/coll.rs [hunk 2]
```diff
@@ -720,3 +723,34 @@ async fn find_one_and_delete_hint_server_version() {
...
+    client
+        .database(function_name!())
+        .collection(function_name!())
+        .find_one(None, options)
+        .await
+        .unwrap();
+
+    let command_started = client.get_successful_command_execution("find").0;
+
+    dbg!(&command_started);
```
Comment (patrickfreed) [inline src/test/coll.rs:754]: this debug print can be removed.

Comment (saghm) [reply src/test/coll.rs:754]: Oops, good catch

**Comment by saghm** [comment]
Hmm, evergreen doesn't seem to be running for this patch now. I'm going to hold off on merging for a bit in case evergreen is just backed up or something.

**Comment by saghm** [comment]
Evergreen is back up and CI passed, so I'll merge now

**Pull request merged**